### PR TITLE
Fixed issue with wrong applying anchors and active

### PR DESCRIPTION
**Pull request by AndriiDovhopolov-okta**
## Description:
- Fixed anchor behavior for sidebar

### Resolves:

* [OKTA-338593](https://oktainc.atlassian.net/browse/OKTA-338593)

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/components/OnThisPageItem.vue [hunk 1]
```diff
@@ -17,31 +17,46 @@ export default {
...
+        return true;
       }
-      let anchor = this.activeAnchor.replace(/^#/, '');
-      return (node.path && node.path == node.basePath + '#' + anchor) || (node.slug && node.slug == anchor);
+      return false;
     },
     setActiveData: function() {
-      this.imActive = this.isActive(this.link);
-      this.iHaveChildrenActive = (this.link.children || [] ).some( child => this.isActive(child) );
+      const currentHash = this.activeAnchor || this.$route.hash
```
Comment (oleksandrpravosudko-okta) [inline packages/@okta/vuepress-theme-prose/components/OnThisPageItem.vue:57]: `currentHash` doesn't seem to be used

File: packages/@okta/vuepress-theme-prose/components/OnThisPageItem.vue [hunk 2]
```diff
@@ -17,31 +17,46 @@ export default {
...
-        return false;
+    calculateLinkHash: function(link){
+      const linkPathDeconstructed = this.link.path.split('/');
+      this.linkHash = linkPathDeconstructed[linkPathDeconstructed.length-1];
+    },
+    isLinkActive: function( node, toLinkHash = null  ) {
+      if(this.activeAnchor && this.activeAnchor === this.linkHash){ 
+        return true 
+      }
+      if(toLinkHash && toLinkHash === this.linkHash) {
```
Comment (oleksandrpravosudko-okta) [inline packages/@okta/vuepress-theme-prose/components/OnThisPageItem.vue:51]: what is the use case for this check? (expecting `this.activeAnchor` check to be sufficient for detecting current hash match)

File: packages/@okta/vuepress-theme-prose/components/OnThisPageItem.vue [hunk 3]
```diff
@@ -17,31 +17,46 @@ export default {
...
+      iHaveChildrenActive: false,
+      linkHash: '',
     }
   },
   components: {
     OnThisPageItem: () => import('../components/OnThisPageItem.vue'),
   },
   mounted() {
     this.setActiveData();
+    this.calculateLinkHash();
```
Comment (oleksandrpravosudko-okta) [inline packages/@okta/vuepress-theme-prose/components/OnThisPageItem.vue:29]: does this need to happen on update as well?
also a nit-pick: `calculate` + `hash` reads like a checksum related operation - perhaps we could use a different name

**Comment by albertchen-okta** [comment]
@AndriiDovhopolov-okta is this PR still valid, needed? What should we do with this?

**Pull request closed without merging**